src/k8s-engine/client.ts

```diff
@@ -166,38 +194,47 @@ export class KubeClient extends events.EventEmitter {
     this.forwarder = new k8s.PortForward(this.kubeconfig, true);
     this.shutdown = false;
     this.coreV1API = this.kubeconfig.makeApiClient(k8s.CoreV1Api);
+    this.services = null;
+  }
 
-    // Set up a watch for services
-    // Since the watch API we have _doesn't_ notify us when things have
-    // changed, we'll need to do some trickery and wrap the underlying watcher
-    // with our own code.
-    class WrappedWatch extends k8s.Watch {
-      callback: ()=>void;
-      constructor(kubeconfig: k8s.KubeConfig, callback:()=>void) {
-        super(kubeconfig);
-        this.callback = callback;
-      }
+  async waitForServiceWatcher() {
+    const startTime = Date.now();
+    const maxWaitTime = 300_000;
+    const waitTime = 3_000;
 
-      watch(
-        path: string,
-        queryParams: any,
-        callback: (phase: string, apiObj: any, watchObj?: any) => void,
-        done: () => void,
-        error: (err: any) => void
-      ): Promise<any> {
-        const wrappedCallback = (phase: string, apiObj: any, watchObj?: any) => {
-          callback(phase, apiObj, watchObj);
-          this.callback();
-        };
+    while (true) {
+      const currentTime = Date.now();
 
-        return super.watch(path, queryParams, wrappedCallback, done, error);
+      if ((currentTime - startTime) > maxWaitTime) {
+        console.log(`Waited more than ${ maxWaitTime / 1000 } secs for kubernetes to fully start up. Giving up.`);
+        break;
+      }
+      if (await this.getServiceListWatch()) {
+        break;
       }
+      await util.promisify(setTimeout)(waitTime);
     }
+  }
 
+  /**
+   * If the `ListWatch` method is called too early, it never fires,
+   * and no services are ever displayed.
```

Comment:
This comment appears to be out of place: it's not a JSDoc comment describing `getServiceListWatch` (what it does, etc.)